db/wal_dump.cc

```diff
@@ -0,0 +1,39 @@
+#include "rocksdb/wal_dump.h"
+#include <memory>
+#include <thread>
+
+#include "rocksdb/env.h"
+#include "db/wal_manager.h"
+#include "db/version_set.h"
+#include "options/db_options.h"
+
+namespace rocksdb {
+
+#ifndef ROCKSDB_LITE
+
+Status WalDump::CreateWalIter(std::string dbname,
+                                   SequenceNumber start_seq,
+                                   SequenceNumber end_seq,
+                                   std::unique_ptr<TransactionLogIterator>* iter,
+                                   const TransactionLogIterator::ReadOptions& read_options) {
+  thread_local ImmutableDBOptions db_options;
```

Comment:
Why thread local for a local variable? Two threads calling this method should have their own stacks.

Reply:
'thread_local' is used to extend the life cycle of variable 'db_options', if it is destructed as a local variable, the created iterator will goes wrong. we can use 'static' too, but it can't be used by multiple threads.
If 'db_options' can be given as a member of class WalDump, it is unnecessary to define as local variable. However, the header of ImmutableDBOptions  is not exposed, if we include it in headers in include/rocksdb/, it will cause compiling error.